Batch improvements (#720)
grafana/k6-docs: src/data/markdown/docs/02 javascript api/08 k6-http/10-batch- requests -.md

| --------- | --------------- | ---------------------------------------------------------------- |
| requests  | array \| object | An array or object containing requests, in string or object form |

> ### Request definition

Reviewer: ```suggestion
```

> 

Reviewer: ```suggestion
```

> #### Array and Object

Reviewer: ```suggestion
```

> A request specified by Array or Object can be defined with the following parameters.

Reviewer: ```suggestion
You can define your request as an array or object with the following parameters.
```

| 3        | body (optional)   | string / object / ArrayBuffer | The body of the request if relevant. Can be set to `null` if not applicable but you want to set the last `params` argument. |
| 4        | params (optional) | object           | [Params](/javascript-api/k6-http/params) like auth, custom headers and tags.                                                |

<Blockquote mod="warning">
When a Request definition is provided as an Array, the order of items is relevant. Check the `Position` column for the correct order.
</Blockquote>

#### String

> If an item is passed as a `string` then it will be automatically parsed as a `GET` request.

Reviewer: I don't understand this. The request definition says that method and URL parameters are required and must be strings.

What information does this section add?

Reviewer: Ah, now I see in the new "Shorthand" example that it looks like the `method` parameter is not required, at least in arrays. I think we need to update the table, right?

Does this shortcut work on objects, too? "Item" makes me think it's only arrays.

Proposed changes:

> As a shortcut, you can pass only the `URL` in arrays. k6 will automatically use `GET` for the request method.

_or, if it can work in an object too_

> If you define only the URL parameter, k6 will automatically use `GET` for the request method.

Reviewer: Oh, I think I see how I got confused.

If I understand correctly, there are really three ways to define batch requests:

- As an array of strings

  ```js
  http.batch(['http://test.k6.io', 'http://test.k6.io/pi.php']);
  ```
- As an array of arrays:
 
   ```js 
    const responses = http.batch([
      ['GET', 'https://test.k6.io', null, { tags: { ctype: 'html' } }],
      ['GET', 'https://test.k6.io/style.css', null, { tags: { ctype: 'css' } }],
      ['GET', 'https://test.k6.io/images/logo.png', null, { tags: { ctype: 'images' } }],
    ]);
    ```
 - As an array of objects
 
    ```js
      const responses = http.batch([req1, req2, req3]);
      ```
 
 Maybe we can mention this information at the beginning. I suppose objects and array batch definitions are simple enough that we can group them in one table, but string should have it's own section.

</CodeGroup>

> _Note that the requests in the example above may happen in any order, or simultaneously. When running requests in batches, there is no guarantee that e.g. req1 will happen before req2 or req3_.

Reviewer: ```suggestion
> **ⓘ Batch requests do not happen in a fixed order.**
> That is, there's no guarantee that `req1` will happen before `req2` or `req3`_.
> In the preceding example, the requests may happen in any order, or simultaneously.
```

### Example with GET shorthand

> If a requests' item is passed as a `string` then they will be automatically parsed as a `GET` request. 

Reviewer: ```suggestion
As a shortcut, you can define `URL` as the only parameter. If you do, k6 will then they will automatically use `GET` as the request method. 
```

> If a requests' item is passed as a `string` then they will be automatically parsed as a `GET` request. 

Reviewer: Again, we should specify whether this works on only arrays, or on arrays and objects.
